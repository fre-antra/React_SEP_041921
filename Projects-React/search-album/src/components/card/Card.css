.card{
    width: 200px;
    text-align: center;
    border-style: solid;
    border-color: var(--color-blue);
    border-width: thin;
    display: flex;
    flex-direction: column;
    margin: 10px;
    padding: 5px;
}
.card img{
    padding: .5rem;
    height: 180px;
}
.card_content{
    font-size: 1.2rem;
    font-weight: 700;
}