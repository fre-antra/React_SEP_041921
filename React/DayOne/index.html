<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <div id="hello-example"></div>
    <script
      crossorigin
      src="https://unpkg.com/react@17/umd/react.development.js"
    ></script>
    <script
      crossorigin
      src="https://unpkg.com/react-dom@17/umd/react-dom.development.js"
    ></script>
    <script>
      class HelloMessage extends React.Component {
        render() {
          return React.createElement('div', null, 'Hello ', this.props.name);
        }
      }

      ReactDOM.render(
        React.createElement(HelloMessage, {
          name: 'Taylor',
        }),
        document.getElementById('hello-example')
      );
    </script>
  </body>
</html>
