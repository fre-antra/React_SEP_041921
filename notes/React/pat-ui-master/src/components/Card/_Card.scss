.card-default {
  border-style: solid;
  border-width: $border-width;
  border-radius: $border-radius;
  border-color: $border-color;

  width: 15rem;
  display: flex;
  flex-direction: column;
  .image {
    height: 10rem;
  }
  .image img {
    height: inherit;
  }
  .body {
    border-style: none;
    padding: 0.6rem;
    margin: 0px;
  }
  .btn {
    width: 10rem;
  }
}

.card-horizontal {
  @include card-border;
  // margin-right: 15px;
  // margin-top: 15px;
  width: 40rem;
  height: 12rem;
  display: flex;
  flex-direction: row;
  .image {
    margin: 0px;
    width: 12rem;
    height: 12rem;
  }
  .image img {
    border-radius: inherit;
    width: 11.8rem;
    height: 11.8rem;
  }
  .body {
    display: flex;
    margin: 0px;
    padding: 10px;
    flex-direction: column;
    justify-content: space-between;
  }
  .btn {
    width: 10rem;
  }
}
.card-circledImage {
  .image {
    height: 10.5rem;
    display: flex;
    border-style: none;
    justify-content: center;
    align-items: center;
  }
  .image img {
    width: fit-content;
    height: fit-content;
    border-radius: 50%;
  }
  .body {
    border-style: none;
    padding: 0.6rem;
    margin: 0px;
    // @include card-body-style;
  }

  .btn {
    width: 10rem;
  }
}

.card-noImage {
  .image {
    margin: 0px;
    display: none;
  }
  .body {
    border-style: none;
    padding: 0.6rem;
    margin: 0px;
  }
  .btn {
    width: 10rem;
  }
}

.card-largeImage {
  .image img {
    border-radius: inherit;
  }
  .body {
    text-align: center;
  }
  .body p {
    display: none;
  }
  .btn {
    display: none;
  }
}
.card-primary {
  border-color: $border-color;
  .body {
    border-color: $border-color;
  }
  .body h5 {
    color: $black;
  }
  .image {
    width: 14.9rem;
    border-style: none;
  }
  .btn {
    @include button-style($white, $primary, $primary);
  }
}
.card-dark {
  border-color: $gray-800;

  .body {
    background-color: $gray-800;
    border-color: $gray-800;
    color: $white;
  }

  .image {
    width: 14.9rem;
    border-style: none;
  }
  .btn {
    @include button-style($white, $secondary, $secondary);
  }
}
.card-purple {
  border-color: $purple;

  .body {
    border-color: $purple;
  }
  .body h5 {
    color: $purple;
  }
  .image {
    width: 14.9rem;
    border-style: none;
  }
  .btn {
    @include button-style($white, $purple, $purple);
  }
}

.card-blue {
  border-color: $blue;

  .body {
    border-color: $blue;
  }
  .body h5 {
    color: $blue;
  }
  .image {
    width: 14.9rem;
    border-style: none;
  }
  .btn {
    @include button-style($white, $blue, $blue);
  }
}

.card-yellow {
  border-color: $dark-yellow;

  .body {
    border-color: $dark-yellow;
  }
  .body h5 {
    color: $dark-yellow;
  }
  .image {
    width: 14.9rem;
    border-style: none;
  }
  .btn {
    @include button-style($white, $dark-yellow, $dark-yellow);
  }
}

.card-sm {
  transform: scale(0.8);
  .image {
    transform: scale(1);
  }
  .body {
    transform: scale(1);
  }
  // .image {
  //   height: fit-content;
  //   width: fit-content;
  //   margin: 0px;
  // }
  // .image img {
  //   height: 7.8rem;
  //   width: 11.8rem;
  // }
  // .body {
  //   font-size: $font-size-sm * 0.8;
  // }
  // .body h5 {
  //   font-size: $font-size-sm;
  // }
  // .btn {
  //   @include button-size(
  //     $btn-padding-y-sm,
  //     $btn-padding-x-sm,
  //     $btn-font-size-sm,
  //     $border-radius-sm
  //   );
  //   width: 8rem;
  // }
}

.card-lg {
  transform: scale(1.2);
  .image {
    transform: scale(1);
  }
  .body {
    transform: scale(1);
  }

  // .body {
  //   font-size: $font-size-base;
  // }
  // .body h5 {
  //   font-size: 1.5rem;
  // }
  // .btn {
  //   width: 12rem;
  // }
}
