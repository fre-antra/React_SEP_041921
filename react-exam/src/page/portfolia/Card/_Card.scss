/*.luvtalk-project, .personal-website-project, .strike-zone-project, .resume {border: 3px solid white;}*/
.card-content{
  min-height: 200px;
}
.luvtalk-project {
    background: linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
      url(../../../assets/images/luvtalklogo.png);
    background-position: center center;
    padding: 50px;
    
  }
  
  .personal-website-project {
    background: linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
      url(../../../assets/images/logo.svg);
    background-position: center center;
    padding: 50px;
    
  }
  
  .strike-zone-project {
    background: linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
      url(../../../assets/images/mlbproject.png);
    background-position: center center;
    padding: 50px;
    
  }