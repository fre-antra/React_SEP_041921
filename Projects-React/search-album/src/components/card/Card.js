import React, {Component} from 'react';
//import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
//import { faMusic } from '@fortawesome/free-solid-svg-icons'
//import {Link} from 'react-router-dom'

import './Card.css'

export default class Card extends Component {
    static defaultProps = {
        album:{}
    }
    render() {
        const {album} = this.props
        return(
            <div className='card'>
                <img src={album.artworkUrl100} alt=''/>
                <div className='card_content'>{album.collectionName}</div>
            </div>
        )
        
    }
}