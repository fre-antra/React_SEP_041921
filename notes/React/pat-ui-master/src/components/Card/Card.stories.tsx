import React from 'react';
import { action } from '@storybook/addon-actions';
import Card from './Card';
import { Icon } from '../../index';

export default {
  title: 'Card',
  component: Card,
};

export const DefaultCard = () => (
  <div>
    <Card
      btnOnClick={action('Default Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        <div>
          <Icon disabled={false} loading={false} name='home' size='tiny' />{' '}
          <span>
            You can put any ReactNode including ReactElement, ReactFragment,
            string, number, array of ReactNodes, null, undefined, boolean in the
            card content{' '}
          </span>
        </div>
      }
      cardTitle={'Default Card'}
      buttonTitle={'show more info'}
    ></Card>
  </div>
);

export const DiffModeCard = () => (
  <div>
    <Card
      btnOnClick={action('Default Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
      cardTitle={'Default Card'}
      buttonTitle={'show more info'}
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Default Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
      cardTitle={'Horizontal Card'}
      buttonTitle={'show more info'}
      cardMode='horizontal'
    ></Card>
  </div>
);

export const DiffTypeCard = () => (
  <div>
    <Card
      btnOnClick={action('Default Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
      cardTitle={'Circle Image Card'}
      buttonTitle={'show more info'}
      cardType='circledImage'
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Default Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
      cardTitle={'No Image Card'}
      buttonTitle={'show more info'}
      cardType='noImage'
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Default Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardTitle={'Large Image Card'}
      cardType='largeImage'
    ></Card>
  </div>
);
export const DiffThemeCard = () => (
  <div>
    <Card
      btnOnClick={action('Dark Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardTitle={'Primary Theme Card'}
      buttonTitle={'show more info'}
      cardTheme='primary'
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Dark Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardTitle={'Dark Theme Card'}
      buttonTitle={'show more info'}
      cardTheme='dark'
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Purple Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardTitle={'Purple Theme Card'}
      buttonTitle={'show more info'}
      cardTheme='purple'
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Yellow Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardTitle={'Yellow Theme Card'}
      buttonTitle={'show more info'}
      cardTheme='yellow'
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
    ></Card>
    <br></br>
    <Card
      btnOnClick={action('Blue Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardTitle={'Blue Theme Card'}
      buttonTitle={'show more info'}
      cardTheme='blue'
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
    ></Card>
  </div>
);

export const DiffSizeCard = () => (
  <div>
    <Card
      btnOnClick={action('Small Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
      cardTitle={'Small Card'}
      cardSize='sm'
      buttonTitle={'show more info'}
    ></Card>
    <br></br>

    <Card
      btnOnClick={action('Large Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        'Some quick example text to build on the card title and make up the bulk of the card content.'
      }
      cardTitle={'Large Card'}
      cardSize='lg'
      buttonTitle={'show more info'}
    ></Card>

    <br></br>
    <Card
      btnOnClick={action('Large Button clicked')}
      cardImgSrc={'https://via.placeholder.com/150'}
      cardParagraph={
        <div>
          <Icon disabled={false} loading={false} name='home' />{' '}
          <span>this is a react node</span>
        </div>
      }
      cardTitle={'Large Card'}
      cardSize='sm'
      buttonTitle={'show more info'}
    ></Card>
  </div>
);
