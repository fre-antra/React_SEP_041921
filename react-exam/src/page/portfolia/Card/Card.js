import React from 'react';

export const Card = (props) => {
  const { style, content } = props;
  return (
    <>
      <div className='col-sm-4 col-md-4 text-center'>
        <div className={style}>
          <div className='card-content'>
            <h3>LUV TALK Website</h3>
            <p>{content}</p>
          </div>

          <div className='d-flex justify-content-center'>
            <div className='project-more-info'>
              <a className='btn btn-default btn-border' href='#'>
                More
              </a>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
