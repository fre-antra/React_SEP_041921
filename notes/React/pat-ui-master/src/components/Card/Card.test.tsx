import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Card, { patCardProps } from './Card';
describe('Card', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Card> Snapshot Card </Card>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render default card', () => {
    const cardProps: patCardProps = {
      btnOnClick: jest.fn(),
    };
    const wrapper = render(<Card {...cardProps}></Card>);
    const cardElement = screen.getByTestId('card-element');
    expect(cardElement).toHaveClass('card card-default');
  });
  it('should render card with different props', () => {
    const cardProps: patCardProps = {
      cardTheme: 'dark',
      cardTitle: 'Dark Card',
      cardSize: 'sm',
      cardType: 'circledImage',
      cardMode: 'horizontal',
      className: 'test',
    };
    const wrapper = render(<Card {...cardProps}></Card>);
    const cardElement = wrapper.queryByText('Dark Card');
    expect(cardElement).toBeInTheDocument();
    const Element = screen.getByTestId('card-element');
    expect(Element).toHaveClass('card card-horizontal card-dark card-sm test');
  });

  it('should render button can be clicked', () => {
    const cardProps = {
      buttonTitle: 'title',
      btnOnClick: jest.fn(),
    };
    //test button
    const wrapper = render(<Card {...cardProps}></Card>);
    const buttonElement = screen.getByTestId('button-element');
    expect(buttonElement.tagName).toBe('BUTTON');
    expect(buttonElement).toHaveTextContent('title');
    expect(cardProps.btnOnClick).toHaveBeenCalledTimes(0);
    fireEvent.click(buttonElement);
    expect(cardProps.btnOnClick).toHaveBeenCalledTimes(1);
  });
  it('should render image with correct source', () => {
    const cardProps = {
      cardImgSrc: 'png',
    };
    const wrapper = render(<Card {...cardProps}></Card>);
    const imageElement = screen.getByTestId('image-element');
    expect(imageElement.tagName).toBe('IMG');
    expect(imageElement.getAttribute('src')).toBe('png');
  });
});
