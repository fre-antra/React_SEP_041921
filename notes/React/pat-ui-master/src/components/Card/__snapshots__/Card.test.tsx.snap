// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card card-default"
    data-testid="card-element"
  >
    <div
      class="card card-default image"
      data-testid="card-image-element"
    >
      <img
        data-testid="image-element"
      />
    </div>
    <div
      class="card card-default body"
      data-testid="card-body-element"
    >
      <h5 />
      <p />
    </div>
  </div>
</DocumentFragment>
`;
